File: codigo/backend/go-backend/models/pedido.go
package models

import "fmt"


type Pedido struct {
    ID         uint    `gorm:"primaryKey"`
    ProdutoID  uint    `gorm:"not null"`
    LojaID     uint    `gorm:"not null"`
    Status     string  `gorm:"size:50;not null"`
    VendedorID uint    `gorm:"not null"`

    Produto  Produto  `gorm:"foreignKey:ProdutoID;references:ID"`
    Loja     Loja     `gorm:"foreignKey:LojaID;references:ID"`
    Vendedor Vendedor `gorm:"foreignKey:VendedorID;references:ID"`
}

func (Pedido) TableName() string {
    return "pedidos"
}

func (p *Pedido) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "id":          p.ID,
        "produto_id":  p.ProdutoID,
        "loja_id":     p.LojaID,
        "status":      p.Status,
        "vendedor_id": p.VendedorID,
    }
}

func (p *Pedido) String() string {
    return fmt.Sprintf("<Pedido %d>", p.ID)
}

File: codigo/backend/sap-ecc-emulator/src/models/pedido.py
from db import db

class Pedido(db.Model):
    __tablename__ = 'pedidos'

    id = db.Column(db.Integer, primary_key=True)
    produto_id = db.Column(db.Integer, db.ForeignKey('produtos.id'), nullable=False)
    loja_id = db.Column(db.Integer, db.ForeignKey('lojas.id'), nullable=False)
    status = db.Column(db.String(50), nullable=False)
    vendedor_id = db.Column(db.Integer, db.ForeignKey('vendedores.id'), nullable=False)

    produto = db.relationship('Produto', backref='pedidos', lazy=True)
    loja = db.relationship('Loja', backref='pedidos', lazy=True)
    vendedor = db.relationship('Vendedor', backref='pedidos', lazy=True)

    def to_dict(self):
        return {
            'id': self.id,
            'produto_id': self.produto_id,
            'loja_id': self.loja_id,
            'status': self.status,
            'vendedor_id': self.vendedor_id
        }

    def __repr__(self):
        return f'<Pedido {self.id}>'
